recipes/test_processes.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: tests_processes
+#
+# Copyright 2013-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+###################
+# supervisord
+###################
+bash 'check supervisord running as root' do
+  cwd Chef::Config[:file_cache_path]
+  code <<-TEST
+      process="supervisord"
+      user="root"
+
+      pgrep -x $process 1>/dev/null
+      if [[ $? != 0 ]]; then
+        >&2 echo "Expected $process to be running"
+        exit 1
+      fi
+
+      pgrep -x $process -u $user 1>/dev/null
+      if [[ $? != 0 ]]; then
+        >&2 echo "Expected $process to be running as $user"
+        exit 1
+      fi
+  TEST
+end
+
+###################
+# clustermgtd
+###################
+if node['cluster']['node_type'] == 'HeadNode' && node['cluster']['scheduler'] == 'slurm'
+  bash 'check clustermgtd running as cluster admin user' do
+    user = node['cluster']['cluster_admin_user']
+    cwd Chef::Config[:file_cache_path]
+    code <<-TEST
+      process="clustermgtd"
```

Comment:
We can create an utility function to be re-used by all the tests, given that the test code seems the same. See libraries/helpers.rb for some examples.